catalog/tests/Catalog_unittest.cpp

```diff
@@ -161,6 +164,19 @@ class CatalogTest : public ::testing::Test {
          ++it1, ++it2) {
       CompareCatalogAttribute(*it1, *it2);
     }
+
+    ASSERT_EQ((expected.index_scheme_ != nullptr), (checked.index_scheme_ != nullptr));
+
+    if (expected.index_scheme_ != nullptr && checked.index_scheme_ != nullptr) {
```

Comment:
Change to `if (checked.index_scheme_ != nullptr) {`, as you've checked it before.